sim-lib/src/random_activity.rs

```diff
@@ -65,3 +69,125 @@ impl Display for NetworkGraphView {
         write!(f, "network graph view with: {} channels", self.nodes.len())
     }
 }
+
+/// PaymentActivityGenerator manages generation of random payments for an individual node.
+pub struct PaymentActivityGenerator {
+    multiplier: f64,
+    expected_payment_amt: u64,
+    source_capacity: u64,
+    event_dist: Exp<f64>,
+}
+
+impl PaymentActivityGenerator {
+    /// Creates a new activity generator for a node, returning an error if:
+    /// - The node has no capacity deployed on the network (we can't send any payemnts).
+    /// - The expected payment amount is more than the source node's capacity (too big to successfully send a payment).
+    pub fn new(
+        source_capacity_msat: u64,
+        expected_payment_amt: u64,
+        multiplier: f64,
+    ) -> Result<Self, SimulationError> {
+        if source_capacity_msat == 0 {
+            return Err(SimulationError::RandomActivityError("".to_string()));
+        }
+
+        if source_capacity_msat < expected_payment_amt {
+            return Err(SimulationError::RandomActivityError(format!(
+						"source capacity: {source_capacity_msat} < expected_payment_amount: {expected_payment_amt}"
+						)));
```

Comment:
This formatting is incredibly ugly to me but this is what `cargo fmt` is delivering to me 💀